Content.Server/Connection/ConnectionManager.IPIntel.cs

```diff
@@ -0,0 +1,377 @@
+﻿using System.Buffers.Binary;
+using System.Net;
+using System.Net.Sockets;
+using System.Threading.Tasks;
+using Content.Server.Database;
+using Content.Shared.CCVar;
+using Content.Shared.Players.PlayTimeTracking;
+using Robust.Shared.Network;
+using Robust.Shared.Utility;
+
+namespace Content.Server.Connection;
+
+// Handles checking/warning if the connecting IP address is sus.
+public sealed partial class ConnectionManager
+{
+    private void InitializeIPIntel()
+    {
+        _cfg.OnValueChanged(CCVars.GameIPIntelEmail, b => _contactEmail = b, true);
+        _cfg.OnValueChanged(CCVars.GameIPIntelBase, b => _baseUrl = b, true);
+        _cfg.OnValueChanged(CCVars.GameIPIntelFlags, b => _flags = b, true);
+        _cfg.OnValueChanged(CCVars.GameIPIntelRejectUnknown, b => _rejectUnknown = b, true);
+        _cfg.OnValueChanged(CCVars.GameIPIntelRejectBad, b => _rejectBad = b, true);
+        _cfg.OnValueChanged(CCVars.GameIPIntelRejectRateLimited, b => _rejectLimited = b, true);
+        _cfg.OnValueChanged(CCVars.GameIPIntelMaxMinute, b => _requestLimitMinute = b, true);
+        _cfg.OnValueChanged(CCVars.GameIPIntelMaxDay, b => _requestLimitDay = b, true);
+        _cfg.OnValueChanged(CCVars.GameIPIntelBackOffSeconds, b => _backoffSeconds = b, true);
+        _cfg.OnValueChanged(CCVars.GameIPIntelBadRating, b => _rating = b, true);
+        _cfg.OnValueChanged(CCVars.GameIPIntelCacheLength, b => _cacheDays = b, true);
+        _cfg.OnValueChanged(CCVars.GameIPIntelExemptPlaytime, b => _exemptPlaytime = b, true);
+        _cfg.OnValueChanged(CCVars.GameIPIntelAlertAdminReject, b => _alertAdminReject = b, true);
+        _cfg.OnValueChanged(CCVars.GameIPIntelAlertAdminWarnRating, b => _alertAdminWarn = b, true);
+    }
+
+    [Dependency] private readonly IHttpClientHolder _http = default!;
+
+    private bool _ratelimitedMinute;
+    private bool _ratelimitedDay;
+    private int _currentRequestsDay;
+    private int _currentRequestsMinute;
+    private TimeSpan _lastRatelimitedDay;
+    private TimeSpan _lastRatelimitedMinute;
```

Comment:
You should honestly move this into a proper struct so the code can be deduplicated between minute/day.